Boost 1.86.0 fails to compile on ppc64le

When trying to compile Boost 1.86.0 on my ppc64le machine, I get the following error:

```

    "g++"  -fvisibility-inlines-hidden -fPIC -m64 -pthread -fvisibility=hidden -pipe -frecord-gcc-switches -Wall -g -O2 -flto=auto -ffat-lto-objects -fno-strict-aliasing  -DBOOST_ALL_NO_LIB=1 -DBOOST_CHARCONV_DYN_LINK=1 -DBOOST_CHARCONV_HAS_QUADMATH -DBOOST_CHARCONV_SOURCE=1 -DBOOST_COBALT_USE_STD_PMR=1 -DNDEBUG   -I"."  -c -o "bin.v2/libs/charconv/build/gcc-13/release/power_64/optimization-off/pch-off/threading-multi/visibility-hidden/to_chars.o" "libs/charconv/build/../src/to_chars.cpp"

In file included from libs/charconv/build/../src/to_chars.cpp:8:
libs/charconv/build/../src/to_chars_float_impl.hpp: In function 'boost::charconv::to_chars_result boost::charconv::detail::to_chars_hex(char*, char*, Real, int)':
libs/charconv/build/../src/to_chars_float_impl.hpp:277:140: error: 'ieee754_binary80' was not declared in this scope; did you mean 'ieee754_binary64'?
  277 |                     typename std::conditional<std::is_same<Real, __float128>::value || BOOST_CHARCONV_LDBL_BITS == 128, ieee754_binary128, ieee754_binary80>::type
      |                                                                                                                                            ^~~~~~~~~~~~~~~~
      |                                                                                                                                            ieee754_binary64
libs/charconv/build/../src/to_chars_float_impl.hpp:277:156: error: template argument 3 is invalid
  277 |                     typename std::conditional<std::is_same<Real, __float128>::value || BOOST_CHARCONV_LDBL_BITS == 128, ieee754_binary128, ieee754_binary80>::type
      |                                                                                                                                                            ^
libs/charconv/build/../src/to_chars_float_impl.hpp:277:157: error: '<declaration error>' is not a template [-fpermissive]
  277 |                     typename std::conditional<std::is_same<Real, __float128>::value || BOOST_CHARCONV_LDBL_BITS == 128, ieee754_binary128, ieee754_binary80>::type
      |                                                                                                                                                             ^~
libs/charconv/build/../src/to_chars_float_impl.hpp:285:13: error: template argument 3 is invalid
  285 |             >::type>::type
      |             ^
libs/charconv/build/../src/to_chars_float_impl.hpp:285:14: error: '<declaration error>' is not a template [-fpermissive]
  285 |             >::type>::type
      |              ^~
libs/charconv/build/../src/to_chars_float_impl.hpp:285:20: error: template argument 3 is invalid
  285 |             >::type>::type
      |                    ^
libs/charconv/build/../src/to_chars_float_impl.hpp:285:21: error: '<declaration error>' is not a template [-fpermissive]
  285 |             >::type>::type
      |                     ^~
libs/charconv/build/../src/to_chars_float_impl.hpp:302:67: error: 'type_layout' has not been declared
  302 |     const Unsigned_Integer denorm_mask = (Unsigned_Integer(1) << (type_layout::significand_bits)) - 1;
      |                                                                   ^~~~~~~~~~~
libs/charconv/build/../src/to_chars_float_impl.hpp:304:61: error: 'type_layout' has not been declared
  304 |     std::uint64_t exponent = extract_exp(value, uint_value, type_layout::significand_bits);
      |                                                             ^~~~~~~~~~~
libs/charconv/build/../src/to_chars_float_impl.hpp:365:33: error: 'type_layout' has not been declared
  365 |         unbiased_exponent = 1 + type_layout::exponent_bias;
      |                                 ^~~~~~~~~~~
libs/charconv/build/../src/to_chars_float_impl.hpp:370:40: error: 'type_layout' has not been declared
  370 |         unbiased_exponent = exponent + type_layout::exponent_bias;
      |                                        ^~~~~~~~~~~
```

--- a/src/to_chars_float_impl.hpp
+++ b/src/to_chars_float_impl.hpp
@@ -274,13 +274,17 @@ to_chars_result to_chars_hex(char* first, char* last, Real value, int precision)
         typename std::conditional<std::is_same<Real, float>::value, ieee754_binary32,
             typename std::conditional<std::is_same<Real, double>::value, ieee754_binary64,
                     #ifdef BOOST_CHARCONV_HAS_FLOAT128
-                    typename std::conditional<std::is_same<Real, __float128>::value || BOOST_CHARCONV_LDBL_BITS == 128, ieee754_binary128, ieee754_binary80>::type
-                    #elif BOOST_CHARCONV_LDBL_BITS == 128
-                    ieee754_binary128
-                    #elif BOOST_CHARCONV_LDBL_BITS == 80
-                    ieee754_binary80
-                    #else
-                    ieee754_binary64
+                    typename std::conditional<std::is_same<Real, __float128>::value, ieee754_binary128,
+                    #endif
+                        #if BOOST_CHARCONV_LDBL_BITS == 128
+                        ieee754_binary128
+                        #elif BOOST_CHARCONV_LDBL_BITS == 80
+                        ieee754_binary80
+                        #else
+                        ieee754_binary64
+                        #endif
+                    #ifdef BOOST_CHARCONV_HAS_FLOAT128
+                    >::type
                     #endif
             >::type>::type
         #ifdef BOOST_CHARCONV_HAS_FLOAT16

